models/pytorch2onnx/ort_run_frozen_each_layer.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import onnxruntime as ort
+import numpy as np
+import argparse
+import os
+import sys
+import time
+import onnx
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--file', type=str, default='./frozen_graph.onnx', help='The file name of the frozen graph.')
+parser.add_argument('--graph_optimization_level', type=str, default='ORT_ENABLE_ALL', help='ONNX Runtime graph optimization level.')
+parser.add_argument('--provider', type=str, default='CPUExecutionProvider', help='The backend provider.')
+parser.add_argument('--logger_severity', type=int, default=2, help='onnxruntime.set_default_logger_severity.')
+args = parser.parse_args()
+
+if not os.path.exists(args.file):
+    parser.exit(1, 'The specified file does not exist: {}'.format(args.file))
+
+onnx.checker.check_model(args.file)
+print("ONNX model check passed!")
+
+def get_numpy(tensor):
+    # ONNX Data Types Doc: https://github.com/onnx/onnx/blob/master/docs/IR.md#standard-data-types
+    # ONNX Data Types Code: https://github.com/onnx/onnx/blob/master/onnx/defs/data_type_utils.h
+    # NumPy Data Types: https://numpy.org/doc/stable/user/basics.types.html
+    def get_numpy_dtype(onnx_dtype):
+        if 'float16' in onnx_dtype:
+            return np.float16
+        elif 'float' in onnx_dtype:
+            return np.float32
+        elif 'double' in onnx_dtype:
+            return np.float64
+        elif 'int8' in onnx_dtype:
+            return np.int8
+        elif 'int16' in onnx_dtype:
+            return np.int16
+        elif 'int32' in onnx_dtype:
+            return np.int32
+        elif 'int64' in onnx_dtype:
+            return np.int64
+        elif 'uint8' in onnx_dtype:
+            return np.uint8
+        elif 'uint16' in onnx_dtype:
+            return np.uint16
+        elif 'bool' in onnx_dtype:
+            return np.bool_
+        else:
+            raise NotImplementedError(onnx_dtype + " is not supported in this script yet.")
+        return np.float32
+
+    dtype = get_numpy_dtype(tensor.type)
+    shape = tensor.shape
+    return np.ones(shape, dtype=dtype)
+
+model = onnx.load(args.file)
+node_name_dict = {}
+for node_id in range(len(model.graph.node)):
+    node_outputs = model.graph.node[node_id].output
+    for output_id in range(len(node_outputs)):
+        debug_tensor = onnx.helper.ValueInfoProto()
+        debug_tensor.name = node_outputs[output_id]
+        model.graph.output.append(debug_tensor)
+        node_name_dict.update({debug_tensor.name: model.graph.node[node_id].name+'_'+str(output_id)})
+onnx.save(model, "/tmp/tmp.onnx")
```

Comment:
Save to local tmp folder or in-memory construct ort session by `ort.InferenceSession(onnx_model.SerializeToString())`?

Reply:
done